src/14_legendre.jl

```diff
@@ -0,0 +1,418 @@
+@doc raw"""
+    chebyshevt(n, x)
+
+Evaluate the Chebyshev polynomials of the first kind ``T_n(x)`` of order ``n`` at position
+``x``, defined by
+
+```math
+T_n(x)
+=
+\begin{cases}
+    1
+    & \text{for} \quad n = 0\,, \; x \in [-1,1]
+    \\
+    x
+    & \text{for} \quad n = 1\,, \; x \in [-1,1]
+    \\
+    2 x T_{n-1}(x) - T_{n-2}(x)
+    & \text{for} \quad n \geq 2\,, \; x \in [-1,1]
+    \,.
+\end{cases}
+```
+
+External links: [DLMF](https://dlmf.nist.gov/18.3.T1),
+[Wikipedia](https://en.wikipedia.org/wiki/Chebyshev_polynomials).
+"""
+function chebyshevt(n::Integer, x::Real)
+    if n < 0
+        throw(DomainError(n, "must be non-negative"))
+    end
+    if x < -1 || x > 1
+        throw(DomainError(x, "must be in the range [-1,1]"))
+    end
+
+    ABC_recurrence(n, x,
+        2, 0, 1,                            # A_m, B_m, C_m
+        y->one(x),                          # P_0(y)
+        y->y)                               # P_1(y)
+end
+
+@doc raw"""
+    chebyshevu(n, x)
+
+Evaluate the Chebyshev polynomials of the second kind ``U_n(x)`` of order ``n`` at position
+``x``, defined by
+
+```math
+U_n(x)
+=
+\begin{cases}
+    1
+    & \text{for} \quad n = 0\,, \; x \in [-1,1]
+    \\
+    2x
+    & \text{for} \quad n = 1\,, \; x \in [-1,1]
+    \\
+    2x U_{n-1}(x) - U_{n-2}(x)
+    & \text{for} \quad n \geq 2\,, \; x \in [-1,1]
+    \,.
+\end{cases}
+```
+
+External links: [DLMF](https://dlmf.nist.gov/18.3.T1),
+[Wikipedia](https://en.wikipedia.org/wiki/Chebyshev_polynomials).
+"""
+function chebyshevu(n::Integer, x::Real)
+    if n < 0
+        throw(DomainError(n, "must be non-negative"))
+    end
+    if x < -1 || x > 1
+        throw(DomainError(x, "must be in the range [-1,1]"))
+    end
+
+    ABC_recurrence(n, x,
+        2, 0, 1,                            # A_m, B_m, C_m
+        y->one(x),                          # P_0(y)
+        y->2y)                              # P_1(y)
+end
+
+@doc raw"""
+    hermiteh(n, x)
+
+Evaluate the (physicists') Hermite polynomials ``H_n(x)`` of order ``n`` at position ``x``,
+defined by
+
+```math
+H_n(x)
+= (-1)^n e^{x^2} \frac{\mathrm{d}^n}{\mathrm{d}x^n} e^{-x^2}
+\quad \text{for} \quad
+x \in \mathbb{R}, \; n = 0, 1, 2, \dots
+```
+
+External links: [DLMF](https://dlmf.nist.gov/18.3.T1),
+[Wikipedia](https://en.wikipedia.org/wiki/Hermite_polynomials).
+"""
+function hermiteh(n::Integer, x::Real)
+    if n < 0
```

Comment:
It's also idiomatic in Base to simply this by writing
`n < 0 && throw(DomainError(n, "must be non-negative"))`

Reply:
done. see commit [956f102](https://github.com/JuliaMath/SpecialFunctions.jl/pull/175/commits/956f10225803adc53b6368c7bcf0db7af484d8c9)